constraint/pkg/externaldata/cache.go

```diff
@@ -69,6 +73,32 @@ func isValidURL(url string) bool {
 	return true
 }
 
+func isValidCABundle(provider *v1alpha1.Provider) error {
+	if provider.Spec.InsecureTLSSkipVerify {
+		if provider.Spec.CABundle != "" {
+			return fmt.Errorf("insecureTLSSkipVerify is set to true but caBundle is not empty")
+		}
+
+		return nil
+	}
+
+	u, err := url.Parse(provider.Spec.URL)
+	if err != nil {
+		return err
+	}
+
+	switch u.Scheme {
+	case "http":
+		return fmt.Errorf("only HTTPS scheme is supported for Providers. To enable HTTP scheme, set insecureTLSSkipVerify to true")
+	case "https":
+		if provider.Spec.CABundle == "" {
```

Comment:
Should we attempt to parse/load the CA bundle? Some checks we do for CA certs can be found in cert controller.